app/routes.py

```diff
@@ -1,2 +1,188 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from flask import Blueprint, request, make_response, jsonify
+from datetime import datetime
+import requests
+import flask_migrate
+import os
+from app.models.goal import Goal
+
+
+
+task_bp = Blueprint("task", __name__, url_prefix='/tasks')
+goal_bp = Blueprint("goal", __name__, url_prefix='/goals')
+
+@task_bp.route("", methods=["POST"], strict_slashes = False)
+def post_task():
+    request_body = request.get_json()
+
+    if "title" in request_body and "description" in request_body and "completed_at" in request_body:
+        new_task = Task(title=request_body["title"],description=request_body["description"], 
+                    completed_at=request_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+        return jsonify({"task": new_task.to_json()}), 201
+    else: 
+        return make_response({"details": "Invalid data"}), 400     
+        
+    
+@task_bp.route("", methods=["GET"], strict_slashes = False)
+def get_all_tasks():
+    sort_method = request.args.get("sort")
+    if sort_method == "asc":
+        tasks = Task.query.order_by(Task.title.asc())            
+    elif sort_method == "desc":
+        tasks = Task.query.order_by(Task.title.desc())   
+    else:
+        tasks = Task.query.all()
+    task_response_body = []  
+    for task in tasks:
+        task_response_body.append(task.to_json())
+    return jsonify(task_response_body), 200
+
+@task_bp.route('/<task_id>', methods=['GET'], strict_slashes = False)
+def get_single_task(task_id):  # same name as parameter route
+    task = Task.query.get(task_id)
+    if not task:
+        return "", 404
+    return make_response({"task": task.to_json()}), 200    
```

Comment:
This is the pattern I was referring to above. First handle the 400 level response, and then get on with the rest of the program. Nice work!